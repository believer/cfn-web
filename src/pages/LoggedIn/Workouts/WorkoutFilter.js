// @flow

import React from 'react'
import type { WOD } from '../../../schema.flow.js'
import styled from 'styled-components'

type Props = {
  selected: WOD[],
  toggleType: (wod: WOD) => void
}

const WorkoutFilterWrap = styled.div`
  align-items: center;
  display: flex;
`

const WorkoutFilterTitle = styled.div`
  font-size: 18px;
  font-weight: 700;
`

const WorkoutWODType = styled.div`
  background-color: ${({ selected, theme, wod }) =>
    selected ? theme[wod] : '#fff'};
  border-radius: 3px;
  box-shadow: 0 3px 3px rgba(0, 0, 0, 0.08);
  color: ${({ selected, inverse }) =>
    selected ? (inverse ? 'inherit' : ' #fff') : 'inherit'};
  cursor: pointer;
  margin-left: 10px;
  padding: 5px 15px;
`

const WorkoutFilter = ({ selected, toggleType }: Props) => {
  return (
    <WorkoutFilterWrap>
      <WorkoutFilterTitle>Filter</WorkoutFilterTitle>
      <WorkoutWODType
        inverse
        selected={selected.includes('DAGENS')}
        onClick={() => toggleType('DAGENS')}
        wod="DAGENS"
      >
        Dagens pass
      </WorkoutWODType>
      <WorkoutWODType
        inverse
        selected={selected.includes('HELG')}
        onClick={() => toggleType('HELG')}
        wod="HELG"
      >
        Helgpass
      </WorkoutWODType>
      <WorkoutWODType
        inverse
        selected={selected.includes('MASTODONT')}
        onClick={() => toggleType('MASTODONT')}
        wod="MASTODONT"
      >
        Mastodontpass
      </WorkoutWODType>
      <WorkoutWODType
        selected={selected.includes('PERFORMANCE')}
        onClick={() => toggleType('PERFORMANCE')}
        wod="PERFORMANCE"
      >
        Performance
      </WorkoutWODType>
      <WorkoutWODType
        selected={selected.includes('FITNESS')}
        onClick={() => toggleType('FITNESS')}
        wod="FITNESS"
      >
        Fitness
      </WorkoutWODType>
      <WorkoutWODType
        selected={selected.includes('TRYOUT')}
        onClick={() => toggleType('TRYOUT')}
        wod="TRYOUT"
      >
        Prova p√•
      </WorkoutWODType>
    </WorkoutFilterWrap>
  )
}

export default WorkoutFilter
