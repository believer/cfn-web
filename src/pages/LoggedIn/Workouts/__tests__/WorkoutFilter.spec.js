// @flow

import React from 'react'
import { mount } from 'enzyme'
import WorkoutFilter from '../WorkoutFilter'
import 'jest-styled-components'

describe('components/WorkoutFilter', () => {
  let component
  let toggleType

  beforeEach(() => {
    toggleType = jest.fn()

    component = mount(
      <WorkoutFilter selected={['DAGENS']} toggleType={toggleType} />
    )
  })

  it('renders WorkoutFilter', () => {
    expect(component).toMatchSnapshot()
  })
})
