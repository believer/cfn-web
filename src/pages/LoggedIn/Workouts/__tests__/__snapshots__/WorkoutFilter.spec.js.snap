// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/WorkoutFilter renders WorkoutFilter 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 {
  font-size: 18px;
  font-weight: 700;
}

.c2 {
  border-radius: 3px;
  box-shadow: 0 3px 3px rgba(0,0,0,0.08);
  color: inherit;
  cursor: pointer;
  margin-left: 10px;
  padding: 5px 15px;
}

.c3 {
  background-color: #fff;
  border-radius: 3px;
  box-shadow: 0 3px 3px rgba(0,0,0,0.08);
  color: inherit;
  cursor: pointer;
  margin-left: 10px;
  padding: 5px 15px;
}

<WorkoutFilter
  selected={
    Array [
      "DAGENS",
    ]
  }
  toggleType={[Function]}
>
  <styled.div>
    <div
      className="c0"
    >
      <styled.div>
        <div
          className="c1"
        >
          Filter
        </div>
      </styled.div>
      <styled.div
        inverse={true}
        onClick={[Function]}
        selected={true}
        wod="DAGENS"
      >
        <div
          className="c2"
          onClick={[Function]}
          selected={true}
        >
          Dagens pass
        </div>
      </styled.div>
      <styled.div
        inverse={true}
        onClick={[Function]}
        selected={false}
        wod="HELG"
      >
        <div
          className="c3"
          onClick={[Function]}
          selected={false}
        >
          Helgpass
        </div>
      </styled.div>
      <styled.div
        inverse={true}
        onClick={[Function]}
        selected={false}
        wod="MASTODONT"
      >
        <div
          className="c3"
          onClick={[Function]}
          selected={false}
        >
          Mastodontpass
        </div>
      </styled.div>
      <styled.div
        onClick={[Function]}
        selected={false}
        wod="PERFORMANCE"
      >
        <div
          className="c3"
          onClick={[Function]}
          selected={false}
        >
          Performance
        </div>
      </styled.div>
      <styled.div
        onClick={[Function]}
        selected={false}
        wod="FITNESS"
      >
        <div
          className="c3"
          onClick={[Function]}
          selected={false}
        >
          Fitness
        </div>
      </styled.div>
      <styled.div
        onClick={[Function]}
        selected={false}
        wod="TRYOUT"
      >
        <div
          className="c3"
          onClick={[Function]}
          selected={false}
        >
          Prova p√•
        </div>
      </styled.div>
    </div>
  </styled.div>
</WorkoutFilter>
`;
